File: occasum-core/src/main/java/pw/stamina/occasum/scan/field/FieldLocatorService.java
package pw.stamina.occasum.scan.field;

import java.lang.reflect.Field;
import java.util.List;

public interface FieldLocatorService {

    List<Field> findAllFields(Object source, FieldIgnorePredicate ignorePredicate);

    static FieldLocatorService standard() {
        return new StandardFieldLocatorService();
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/scan/field/FieldLocatorService.kt
package pw.stamina.occasum.scan.field

import java.lang.reflect.Field

interface FieldLocatorService {

    fun findAllFields(source: Any, ignorePredicate: FieldIgnorePredicate): Collection<Field>

    companion object {

        fun standard(): FieldLocatorService {
            return StandardFieldLocatorService()
        }
    }
}
